### PR TITLE
Fix: chunk string error

### PR DESCRIPTION
**Pull request by wongoo**
**What this PR does**:
1. fix that may exceed byte buffer for special chunk string
2. fix that check EOF error as string end, should check reset buffer length instead

### CODE REVIEW

File: encode_test.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func javaDecodeValidate(method string, target interface{}) (string, error) {
 
 	stdin, err := cmd.StdinPipe()
 	if err != nil {
-		fmt.Printf("call java error: %v", err)
+		fmt.Printf("call java error: %v\n", err)
```
Comment (AlexStocks) [inline encode_test.go]: u should use t.Logf in go test

File: encode_test.go [hunk 2]
```diff
@@ -57,7 +57,9 @@ func javaDecodeValidate(method string, target interface{}) (string, error) {
 	}()
 
 	out, err := cmd.Output()
+
 	if err != nil {
+		fmt.Printf("get java result error: %v\n", err)
```
Comment (AlexStocks) [inline encode_test.go]: t.Logf

File: encode_test.go (hunk 1 above)
Comment (wongoo) [reply encode_test.go]: there is not a testing.T parameter for the method

**Comment by fangyincheng** [comment]
I've already tested it.  LGTM.

File: encode_test.go (hunk 1 above)
Comment (AlexStocks) [reply encode_test.go]: this is not a right excuse. this func is invoked by testJavaDecode. so i think u can add a testing.T as this func's parameter.

Comment (wongoo) [reply encode_test.go]: yes, u r correct

**Pull request merged**